converse.js

```diff
@@ -593,8 +596,10 @@
         this.reconnect = function (condition) {
             converse.log('Attempting to reconnect in 5 seconds');
             converse.giveFeedback(__('Attempting to reconnect in 5 seconds'), 'error');
-            setTimeout(function () {
+            clearTimeout(converse.reconnectTimeout);
+            converse.reconnectTimeout = setTimeout(function () {
                 if (converse.authentication !== "prebind") {
+                    this.connection.pass = this.password;
                     this.connection.connect(
```

Comment:
Yeap, sorry for that! When using a token as a password, maybe tokens are valid one time, if reconnects an external funcion update token and set to this.password variable, my mistake was not think about it can be without a pass.